smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -382,7 +382,8 @@ private void writeHeaders(
                                         String headerValue = getInputValue(context, binding.getLocation(),
                                                 memberLocation + "![suffix]", binding.getMember(), target);
                                         // Append the prefix to key.
-                                        writer.write("acc[$S + suffix] = $L;", binding.getLocationName(), headerValue);
+                                        writer.write("acc[$S + suffix!.toLowerCase()] = $L;",
+                                                binding.getLocationName().toLowerCase(Locale.US), headerValue);
```

Comment:
nit: suffix is defined in this block so the [TypeScript's definite assignment assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#definite-assignment-assertions) is not required.
```suggestion
                                        writer.write("acc[$S + suffix.toLowerCase()] = $L;",
                                                binding.getLocationName().toLowerCase(Locale.US), headerValue);
```